datalad/support/gitrepo.py

```diff
@@ -3444,6 +3468,49 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
         # TODO yield result for commit, prev helper checked hexsha pre
         # and post...
 
+    def _check_for_win_compat(self, files):
+        """Check file names for illegal characters or reserved names on Windows
+
+        In the case that a non-Windows-compatible file is detected, warn users
+        about potential interoperability issues.
+        """
+        from collections import defaultdict
+        problems = defaultdict(list)
+        for file in files.keys():
```

Comment:
assuming files is a dict, just
```suggestion
        for file in files:
```
would make it to follow "generic programming" principle and make code more "generic" (e.g. files could be a list or a dict and still work)